src/Emulator/Extensions/Mocks/HPSHostController.cs

```diff
@@ -171,6 +174,114 @@ private byte[] GetBytesFromSlave(int count, TimeInterval timeInterval)
             return result;
         }
 
+        private void PollForError(TimeInterval timeInterval){
+            var errors = ReadCommonErrorStatusBytes(timeInterval);
+            var sw = new Stopwatch();
+            sw.Start();
+            while (errors[0] == 0x00 && sw.Elapsed < new TimeSpan(0,0,3)){
+                errors = ReadCommonErrorStatusBytes(timeInterval);
+            }
+        }
+
+        // When we need to read for more than 255 bytes, we will need this method
+        private byte[] GetLongBytes(int count, TimeInterval timeInterval) {            
```

Comment:
I'm unsure about GetLongBytes... I'm also unsure about the existing GetBytesFromSlave method. Our firmware expects to receive a start-read event then it will send out data for as long as the controller keeps requesting bytes. I don't see anything that does the start-read event. It seems like there's just reading of bytes without first having a start event